luxjs/forms/utils.js

```diff
@@ -3,54 +3,290 @@
+                    // Extend current options
+                    options = scope[target.name];
+
+                    if (options.length > config.optionsTotal) {
+                        defer.reject();
+                        return defer.promise;
+                    } else {
+                        // Increase offset
+                        config.params.offset += config.params.limit;
+                    }
+                } else {
+                    // Add initial options
+                    var options = [];
+                    scope[target.name] = options;
+
+                    config.initialValue.id = '';
+                    config.initialValue.name = 'Loading...';
+
+                    options.push(config.initialValue);
+                }
+
+                if (searchValue === undefined)
+                    delete config.params[config.id];
+                else
+                    config.params[config.id] = searchValue;
 
-    .directive('remoteOptions', ['$lux', function ($lux) {
+                api.get(null, config.params).then(function(data) {
+                    // Get amount of total items
+                    config.optionsTotal = data.data.total;
+
+                    if (searchValue !== undefined && data.data.result.length === 0) {
+                        options[0].name = 'Cannot find value';
+                    } else {
+                        options[0].name = 'Please select...';
+                    }
+
+                    angular.forEach(data.data.result, function (val) {
+                        var name;
+                        if (config.nameFromFormat) {
+                            name = formatString(config.nameOpts.source, val);
+                        } else {
+                            name = val[config.nameOpts.source];
+                        }
+
+                        var optionId;
+                        if (attrs.multiple) {
+                            // For multiple field always get id of the object
+                            optionId = val.id;
+                        } else {
+                            optionId = val[config.id];
+                        }
+
+                        options.push({
+                            id: optionId,
+                            name: name
+                        });
+                    });
+
+                }, function(data) {
+                    options[0] = '(error loading options)';
+                    defer.reject();
+                });
+                return defer.promise;
+            },
+            //
+            setupQueryInitial: function(config) {
+                config.params.limit = config.queryInitial.limit;
+                config.params.offset = config.queryInitial.offset;
+                delete config.params[config.id];
+            }
+        };
+    }])
+    //
+    .directive('selectInfinity', ['$parse', '$timeout', function($parse, $timeout) {
+        function height(elem) {
```

Comment:
Is this the best place for these functions? Can we write unit tests for them if they're here?